warehouse/models/intermediate/transit_database/int_transit_database__urls_to_gtfs_datasets.sql

```diff
@@ -1,21 +1,44 @@
-WITH
+{{ config(materialized='table') }}
 
-gtfs_datasets AS (
-    SELECT * FROM {{ ref('stg_transit_database__gtfs_datasets') }}
+-- TODO: add handling for back-dating mappings before the GTFS dataset record was created
+
+WITH gtfs_datasets AS (
+    SELECT *
+    FROM {{ ref('dim_gtfs_datasets') }}
+    WHERE base64_url IS NOT NULL
 ),
 
-int_transit_database__urls_to_gtfs_datasets AS (
+appearance_duration AS (
     SELECT
         base64_url,
-        key AS gtfs_dataset_key -- this is defined in {{ ref('dim_gtfs_datasets') }}
+        MAX(_valid_to) AS latest_app,
+        MIN(_valid_from) AS first_app
     FROM gtfs_datasets
-    WHERE base64_url IS NOT NULL
-    -- NOTE: there could be more than 1 record per URL per ts if a given
-    -- URL is set in more than one gtfs_dataset row; our general assertion
-    -- is that we really should think of the underlying Airtable data
-    -- as unique at the URL level and merge records if necessary to
-    -- enforce that uniqueness constraint
-    QUALIFY ROW_NUMBER() OVER (PARTITION BY base64_url ORDER BY ts DESC) = 1
+    GROUP BY base64_url
+),
+
+int_transit_database__urls_to_gtfs_datasets AS (
+    SELECT
+        gtfs_datasets.base64_url,
+        gtfs_datasets.original_record_id,
+        gtfs_datasets.key AS gtfs_dataset_key,
+        CASE
+            WHEN gtfs_datasets._valid_from = appearance_duration.first_app THEN CAST('1900-01-01' AS TIMESTAMP)
+            ELSE gtfs_datasets._valid_from
+        END AS _valid_from,
+        gtfs_datasets._valid_to
+    FROM gtfs_datasets
+    LEFT JOIN appearance_duration
+        USING (base64_url)
```

Comment:
I'm surprised sqlfluff didn't alert on this

Reply:
why?

Reply:
oh I forgot we'd disabled https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.rules.Rule_L032